### PR TITLE
Refactor/profile-page

### PR DESCRIPTION
**Pull request by sajald77**
resolves GYS-6336, GYS-6335, GYS-6305, GYS-6269, GYS-6181, GYS-6308, GYS-6304, GYS-6264, GYS-4027, GYS-6340, GYS-6309, GYS-6155

### CODE REVIEW

File: src/graphql/fragments/userContribution.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const FRAGMENT_USER_PROJECT_CONTRIBUTIONS = gql`
+  fragment UserProjectContributions on UserProjectContribution {
+    project {
+      id
+      thumbnailImage
+      name
+      title
+    }
+    funder {
```
[the remark this answers is not in the document]
Comment (steliosrammos) [reply src/graphql/fragments/userContribution.ts:11]: https://linear.app/geyser/issue/GYS-6383/implement-pagination-on-funderfundingtxs-field-for-user-profile

**Review by steliosrammos — CHANGES_REQUESTED** [review]
Big claenup PR! Good work @sajald77 💪 it's nice to see Jotai in action too, I love the clean API it has. 

I dropped a couple comments, but overall looks great.

File: src/graphql/fragments/userContribution.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { gql } from '@apollo/client'
+
+export const FRAGMENT_USER_PROJECT_CONTRIBUTIONS = gql`
+  fragment UserProjectContributions on UserProjectContribution {
+    project {
+      id
```
[the remark this answers is not in the document]
Comment (sajald77) [reply src/graphql/fragments/userContribution.ts:6]: yep, good catch

File: src/graphql/fragments/userContribution.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sajald77) [reply src/graphql/fragments/userContribution.ts:11]: Adding the filter to remove podcast keysends

File: src/pages/profile/views/profileTabs/ProfileTabs.tsx
[the remark this answers is not in the document]
Comment (sajald77) [reply src/pages/profile/views/profileTabs/ProfileTabs.tsx:1]: Yes that is a good point, these are basically ready for being part of the module.

File: src/pages/profile/views/profileTabs/components/TabComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  sub?: string | number
+  isLoading?: boolean
+  renderComponent: JSX.Element
+}
+
+export const TabComponent = ({ tabs, isLoading }: TabComponentProps) => {
+  const { t } = useTranslation()
+  const isMobile = useMobileMode()
+  return (
+    <Tabs id={ID.profile.tabs} display="flex" flexDirection="column" h="full" variant="unstyled" size="sm" w="full">
```
[the remark this answers is not in the document]
Comment (sajald77) [reply src/pages/profile/views/profileTabs/components/TabComponent.tsx:25]: ID is a constant with a list of id's for different parts of the platform, in a nested object like fashion. 
Yeah, we use all caps for constants, it's pretty straight forward , and short.  
I could change it to ELEMENT_IDS, but that's only marginally better in-terms of being descriptive. I can add some comments in the file with ID, to make it more clear.

**Review by steliosrammos — APPROVED** [review]
lgtm!

**Pull request merged**